js/gestionPresupuestoWeb.js

```diff
@@ -101,11 +164,146 @@ function mostrarGastosAgrupadosWeb(idElemento, agrup, periodo) {
+    
+    let fecha =prompt("introduzca la fecha del gasto en formato yyyy-mm-dd");
+    
+    let etiqueta = prompt("introduzca las etiquetas separadas por comas");
+    
+    let etiquetas = etiqueta.split(',');
+    
+    const gasto1 = new gestion.CrearGasto(descripcion, valor, fecha, etiquetas);
+    gestion.anyadirGasto(gasto1);
+    let viejoPresupuesto = document.getElementById('presupuesto');
+    viejoPresupuesto.remove();
+    let viejoGasto = document.getElementById('gastos-totales');
+    viejoGasto.remove();
+    let viejoBalance = document.getElementById('balance-total');
+    viejoBalance.remove();
+    
+    repintar();
+    
+
+
+}
+
+function editarHandle() {
+    this.handleEvent = function(gasto){
+        // this.datosGasto = gasto;
+        let viejoPresupuesto = document.getElementById('presupuesto');
+        viejoPresupuesto.remove();
+        let viejoGasto = document.getElementById('gastos-totales');
+        viejoGasto.remove();
+        let viejoBalance = document.getElementById('balance-total');
+        viejoBalance.remove();
+
+        gasto = this.datosGasto;
+        console.log("EditarHandle : " +this.datosGasto);
+        
+        let nuevaDescripcion = prompt("introduzca la descripcion del gasto",this.datosGasto.descripcion);
+        let nuevoValor = Number(prompt("introduzca el importe del gasto",this.datosGasto.valor),10);
+        
+        let nuevaFecha =prompt("introduzca la fecha del gasto en formato yyyy-mm-dd",this.datosGasto.fecha);
+        
+        let nuevasEtiquetas = prompt("introduzca las etiquetas separadas por comas",this.datosGasto.etiquetas);
+        
+        
+
+        //actualizamos las propiedades del gasto utilizando las funciones correspondientes
+        gasto.actualizarDescripcion(nuevaDescripcion);
+        gasto.actualizarValor(nuevoValor);
+        gasto.actualizarFecha(nuevaFecha);
+        if(nuevasEtiquetas !=""){
+            gasto.anyadirEtiquetas(nuevasEtiquetas.split(','));
+        }
+        
+        
+        repintar();
+    }
+} 
+function BorrarHandle(){
+    this.handleEvent = function(gasto){
+        let viejoPresupuesto = document.getElementById('presupuesto');
+        viejoPresupuesto.remove();
+        let viejoGasto = document.getElementById('gastos-totales');
+        viejoGasto.remove();
+        let viejoBalance = document.getElementById('balance-total');
+        viejoBalance.remove();
+        // this.datosGasto = gasto;
+        gasto = this.datosGasto;
+        console.log("BorrarHandle : " +this.datosGasto);
+        gestion.borrarGasto(gasto.id);
+        repintar();
+    }
+}
+
+function BorrarEtiquetasHandle(){
+    this.handleEvent = function(gasto,etiqueta){
+        console.log("se ha pulsado borrar etiquetas");
+        gasto = this.datosGasto;
+        // this.gasto = gasto;
+        etiqueta = this.datosEtiqueta;
+        // this.etiqueta = etiqueta;
+        
+        gasto.borrarEtiquetas(etiqueta);
```

Comment:
Puedes llamar directamente a `this.gasto.borrarEtiquetas(this.etiqueta)`. Lo mismo en el manejador de borrar.